src/plugins/expressions/common/execution/execution.ts

```diff
@@ -454,9 +468,30 @@ export class Execution<
     input: unknown,
     args: Record<string, unknown>
   ): Observable<UnwrapReturnType<Fn['fn']>> {
+    let hash: string | undefined;
+
     return of(input).pipe(
       map((currentInput) => this.cast(currentInput, fn.inputTypes)),
-      switchMap((normalizedInput) => of(fn.fn(normalizedInput, args, this.context))),
+      map((normalizedInput) => {
+        if (fn.allowCache && this.context.allowCache) {
+          hash = calculateObjectHash([
```

Comment:
I don't have a great knowledge of Rxjs but are we overwriting the hash each time (if the `input` is an array) and using only the last one as the actual hash? shouldn't we pass down the pipe the computed hash?

Reply:
input is an observable. Each value emited by the observable is going to get mapped by this function and then the next map.  whenever we get new input we need to recalculate the hash.